src/efs_parser/BaseFile.py

```diff
@@ -135,13 +145,21 @@ def get_bytes_http(self, offset, size):
                 # connection redirected and found resource - usually https
                 new_loc = response.getheader("Location")
                 # print("url redirected & found ", new_loc)
-                self.parse_url(new_loc)    
+                self.parse_url(new_loc)
                 self.conn.request("GET", url=self.fuparse.path, headers=headers)
-                response = self.conn.getresponse()    
-                resp = response.read()    
+                response = self.conn.getresponse()
+                resp = response.read()
             else:
                 resp = response.read()
-            return resp[:size]       
+            return resp[:size]
+
+    def get_bytes_from_s3(self, offset, size):
+        s3client = boto3.client('s3', region_name='us-west-2', config=Config(signature_version=UNSIGNED))
+        key = self.file.replace(f"s3://{self.bucketname}/","")
+        bytes_range = "bytes=%d-%d" % (offset, offset+size)
+        resp = s3client.get_object(Bucket=self.bucketname,Key=key,Range=bytes_range)
+        data = resp['Body'].read()
+        return data[:size]
 
```

Comment:
this seems to be very specifically written for the public aws encode bucket ...

The idea is I would give you a s3 URL; you could customize this to the format you want 

e.g.`s3://bucket_name@s3_region`

then you split by `@` to get the url and the region to create a client and make the necessary calls for range requests.

This needs to be generic and not specific

Reply:
Done

Reply:
any reason why this has to be this way and not take advantage of s3 url structure, instead of  `s3://bucket_name@s3_region/<file_name>` why not `<s3URL>@<s3_region>` ?

`s3URL` will always be in the following structure `s3://<bucket_name>/<file_name>`, I would rather not break this structure and make it complicated to introduce region in there. It will also be hard for us to generate these urls if this is the case. most times, public buckets give us a s3 url and a region. So a natural thing to do would be`<s3URL>@<s3_region>`

Reply:
unless there's a strong reason on why we should introduce region with the s3 url

Reply:
There is no reason, I just followed your example
e.g.s3://bucket_name@s3_region

Reply:
I will rewrite the url to follow the following structure
s3://<bucket_name>/<file_name>@region
Is this ok?

Reply:
yup `<s3URL>@<s3_region>` should work

Reply:
Done